libraries/HumanPoseFuser.js

```diff
@@ -526,16 +686,242 @@ class HumanPoseFuser {
+     */
+    triangulatePoint(projectionMatrices, points2D, weights) {
+
+        let point3D = [];
+
+        try {  // to catch any exception from mljs lib
+            let AArr = [];  // row by row
+            for (let v = 0; v < projectionMatrices.length; v++) {
+                const P = projectionMatrices[v];
+
+                AArr.push(Matrix.sub(P.getRowVector(2).mul(points2D[v][0]), P.getRowVector(0)).mul(weights[v]).getRow(0));
+                AArr.push(Matrix.sub(P.getRowVector(2).mul(points2D[v][1]), P.getRowVector(1)).mul(weights[v]).getRow(0));
+            }
+
+            // solve homogeneous system A*x = 0
+            // set options to save computation
+            const svdOptions = {
+                computeLeftSingularVectors: false,
+                computeRightSingularVectors: true,
+                autoTranspose: false
+            };
+            let A = new Matrix(AArr);
+            let svd = new SingularValueDecomposition(A, svdOptions);
+
+            if (svd.rank < A.columns) {
+                // cannot get 3D point from under-constrained system
+                return null;
+            }
+
+            // search for the smallest singular value to pick one of right singular vectors
+            let singularValues = svd.diagonal; // Array
+            let orderedSingularValues = [];
+            for (let i = 0; i < singularValues.length; i++) {
+                orderedSingularValues.push([i, singularValues[i]]);
+            }
+            orderedSingularValues.sort((a, b) => a[1] - b[1]); // in ascending order
+            let x = svd.rightSingularVectors.getColumnVector(orderedSingularValues[0][0]); // select the vector for the smallest singular value
+            let nx = Matrix.div(x, x.get(3, 0));  // normalise to get 3D position
+
+            // NOTE: can singular value and especially related singular vector a zero vector?
+            if (Math.abs(orderedSingularValues[0][1]) < svd.threshold) {
+                console.warn(`Near-zero singular value: ${orderedSingularValues[0][1]}; x = ${x}; nx = ${nx}`);
+            }
+
+            point3D = [nx.get(0, 0), nx.get(1, 0), nx.get(2, 0)];
+
+        } catch (error) {
+            console.warn('Point triangulation: ' + error);
+            return null;
+        }
+
+        return point3D;
+    }
+
+    /** Triangulates a set of poses associated with a fused human object.
+     * @param {Array.< [string, WholePoseData] >} poseDataArr - current poses for human objects associated with the fused object
+     * @return {WholePoseData | null} triangulated 3D pose or null when failed
+     */
+    triangulatePose(poseDataArr) {
+        if (poseDataArr.length < 2) {
+            return null;
+        }
+
+        // prepare data
+        let projectionMatrices = []; // per-view
+        let jointsInView = []; // joint 2D positions + confidences per view
+        let latestTS = 0; // the latest data timestamp
+        for (let [id, pose] of poseDataArr) {
```

Comment:
another nit inline, could be _id to make it clear that you don't need the id from poseDataArr's entries